src/Fields/Field.php

```diff
@@ -392,6 +441,52 @@ public function exceptOnForms(): self
         return $this;
     }
 
+    /**
+     * Set the validation rules for the field.
+     */
+    public function rules(string | array | Rule $rules): self
+    {
+        $this->rules = $rules instanceof Rule || is_string($rules) ? func_get_args() : $rules;
+
+        return $this;
+    }
+
+    /**
+     * Set the validation rules for the field.
+     */
+    public function creationRules(string | array | Rule $rules): self
+    {
+        $this->creationRules = $rules instanceof Rule || is_string($rules) ? func_get_args() : $rules;
+
+        return $this;
+    }
+
+    /**
+     * Determine the validation rules for the field in the time of creation.
+     */
+    protected function getCreationRules(): array
+    {
+        return array_merge_recursive($this->rules, $this->creationRules);
+    }
```

Comment:
This is where validation + declarative syntax becomes a bit tricky, so bear with me please 😀 I'm not sure about using `creationRules()` and `updateRules()`. Why? Because people also have other operations that use Backpack fields. And this month (or next month the latest), we'll share tutorials and add a feature to DevTools to easily create operations that use Backpack forms (eg. Moderate, Reply, Ship, etc.). If `creationRules()` and `updateRules()` are hard-coded... it means there's no way devs can use this Resource syntax on custom operations, which... is not the end of the world, but... it's a shame.

Then again, the alternative isn't great either. 

---

In Backpack v3.x we were pointing to the operation where those rules apply:

```php
$this->crud->setFormRequest(ArticleRequest::class, 'create');
$this->crud->setFormRequest(ArticleRequest::class, 'update');
```

That was because we thought it would be easy to just define everything in one place (in Backpack's case, in the `setup()` method). But after working on complex projects... it turns out for medium-complexity and high-complexity projects, you probably want to define each operation individually. So we came up with `setupListOperation()`, `setupCreateOperation()` and `setupUpdateOperation()`.

Ok so that's the history 😅 What can we do here?

---

If this is indeed a problem worth solving (debatable), I see two options:

(A) magic methods that follow a convention, starting from the operation name (eg. if you do `updateRules()` or `createRules()` or `replyRules()` even though that method doesn't exist, it will work; downside: magic, no IDE auto-completion;

(B) extra parameter for the `rules()` method, eg. 
- `->rules('required|min:5', 'create');`
- `->rules('required|min:5', ['create', 'update']);`
- `->rules('required|min:5'); // if missing it can default to both create and update`